contrib/buildrpm/buildrpmLibfabric.sh

```diff
@@ -0,0 +1,332 @@
+         ;;
+      d) configure_options="$configure_options --enable-debug"
+         ;;
+      c) configure_options="$configure_options $OPTARG"
+         ;;
+      r) configure_options="$rpmbuild_options $OPTARG"
+         ;;
+      s) unpack_spec="true"
+         ;;
+      v) verbose="true"
+         ;;
+      h) echo "$usage"
+         exit 0
+         ;;
+    esac
+done
+shift $(( OPTIND - 1 ));
+
+if [[ -n "$verbose" ]]; then
+  verboseoption="-v"
+fi
+
+########################################
+# Check if there is at least 1 parameter
+# specified (tarball is mandatory)
+########################################
+if [[ $# -lt 1 ]]; then
+  echo "$usage" 1>&2
+  exit 1
+fi
+
+#############
+# Print hello
+#############
+if [[ -n "$verbose" ]]; then
+cat <<EOF
+
+------------------------------------------------------------------------------
+====              Welcome to the Libfabric RPM build script               ====
+------------------------------------------------------------------------------
+You have specified these options:
+ $arguments
+------------------------------------------------------------------------------
+
+EOF
+  if [[ -n "$noop" ]]; then
+    echo "$0: --> Option -n was specified. I won't do anything."
+  fi
+fi
+
+################################
+# Check if tarball was specified
+################################
+tarball="$1"
+
+if [[ ! -r "$tarball" ]]; then
+  error 2 "Can't find $tarball"
+fi
+verbose "Found tarball: $tarball"
+
+##############################
+# Get tarball name and version
+##############################
+tardirname=$(basename "$tarball" | awk '{print substr ($0, 0, index($0, ".tar")-1)}')
+if [[ -z "$tardirname" ]]; then
+  error 3 "Cannot determine name from $tarball"
+fi
+verbose "Name of package: $tardirname"
+
+version=$(basename "$tardirname" | cut -d- -f2)
+if [[ -z "$tardirname" ]]; then
+  error 4 "Cannot determine version from $tarball"
+fi
+verbose "Version: $version"
+
+if [[ -n "$install_in_opt" ]]; then
+  prefix="/opt/libfabric/$version"
+  rpmbuild_options="$rpmbuild_options --define '_prefix $prefix'"
+  verbose "Setting RPM install path to: $prefix"
+fi
+
+######################################
+# Try to unpack spec file from tarball
+######################################
+if [[ -f "$specfile" ]]; then
+  if [[ -n "$verbose" ]]; then
+    echo "$0: --> WARNING: $specfile already exists and will be overwritten" 1>&2
+  fi
+fi
+
+if [[ -n "$unpack_spec" ]]; then
+  verbose "Extracting tarball"
+  runcmd "tar $verboseoption -xf \"$tarball\""
+  verbose "Copying specfile"
+  runcmd "cp -fp $verboseoption \"$tardirname/$specfile\" ."
+  verbose "Cleanup after extraction"
+  runcmd "rm -rf $verboseoption \"$tardirname\""
+fi
+
+##############################
+# Check if specfile is present
+##############################
+if [[ ! -r $specfile ]]; then
+  if [[ -z "$noop" ]]; then
+    error 5 "Cannot find $specfile"
+  else
+    verbose "WARNING: Cannot find $specfile"
```

Comment:
Above, you emit a warning to stderr, but here, you do not.  Should you make a `warn()` subroutine, analogous to `verbose()` and `error()`?